teos/src/api/internal.rs

```diff
@@ -402,6 +442,99 @@ mod tests_private_api {
         ));
     }
 
+    #[tokio::test]
+    async fn test_get_appointments() {
+        let internal_api = create_api().await;
+
+        let locator = generate_dummy_appointment(None).locator().serialize();
+        let response = internal_api
+            .get_appointments(Request::new(msgs::GetAppointmentsRequest { locator }))
+            .await
+            .unwrap()
+            .into_inner();
+
+        assert!(matches!(response, msgs::GetAppointmentsResponse { .. }));
+    }
+
+    #[tokio::test]
+    async fn test_get_appointments_watcher() {
```

Comment:
The only thing I really see missing (both in here and in `test_get_appointments_responder`) is testing that more than one appointment is returned if there is more than one appointment with the same locator. This could be used to the existing tests (e.g. modify the test so `i` appointments are added instead of always just one)